packages/store-sync/src/query-cache/recs/Query.ts

```diff
@@ -0,0 +1,141 @@
+import { Entity, QueryFragmentType } from "@latticexyz/recs";
+import { QueryCacheStore } from "../createStore";
+import { query } from "../query";
+import { SchemaToPrimitives, Table } from "@latticexyz/store";
+import { KeySchema, SchemaToPrimitives as SchemaToPrimitivesProtocol } from "@latticexyz/protocol-parser";
+import { encodeEntity } from "../../recs";
+import { hexToResource } from "@latticexyz/common";
+import { QuerySubjects } from "../common";
+import { SubjectRecords } from "@latticexyz/query";
+
+type HasQueryFragment<T extends Table> = {
+  type: QueryFragmentType.Has;
+  table: T;
+};
+
+type NotQueryFragment<T extends Table> = {
+  type: QueryFragmentType.Not;
+  table: T;
+};
+
+type HasValueQueryFragment<T extends Table> = {
+  type: QueryFragmentType.HasValue;
+  table: T;
+  value: SchemaToPrimitives<T["valueSchema"]>;
```

Comment:
should switch these to `schema`

Reply:
interestinggg. really differs from RECS but seems better?